src/main.cpp

```diff
@@ -3041,13 +3042,13 @@ void UnlinkPrunedFiles(std::set<int>& setFilesToPrune)
 }
 
 /* Calculate the block/rev files that should be deleted to remain under target*/
-void FindFilesToPrune(std::set<int>& setFilesToPrune)
+void FindFilesToPrune(std::set<int>& setFilesToPrune, uint64_t nPruneAfterHeight)
```

Comment:
It seems to me that nPruneAfterHeight is precisely the only "implementation detail" from CChainParams that FlushStateToDisk() needs to know about.